components/sup/src/main.rs

```diff
@@ -289,7 +285,7 @@ async fn split_apart_sup_run(sup_run: SupRun,
                         organization: sup_run.organization,
                         gossip_permanent: sup_run.permanent_peer,
                         ring_key,
-                        gossip_peers: sup_run.peer,
+                        gossip_peers: sup_run.peer.iter().map(|x| x.into()).collect(),
```

Comment:
You could get rid of the clippy lint by using this form:

```rust
gossip_peers: sup_run.peer.iter().map(Into::into).collect()
```

It basically uses the function `into` of the `Into<T>` trait as the mapping function itself. In this case the T would be automatically inferred by the compiler due to the assignment to gossip_peers. If it isn't able to infer it you would have to use a slightly modified form:



```rust
gossip_peers: sup_run.peer.iter().map(Into::<SocketAddr>::into).collect()
```

Reply:
It was able to infer by just using (Into::into). I have re-confirmed the same by necessary debugging logs.